src/main/java/de/dennisguse/opentracks/publicapi/CreateMarkerActivity.java

```diff
@@ -0,0 +1,69 @@
+package de.dennisguse.opentracks.publicapi;
+
+import android.content.Intent;
+import android.location.Location;
+import android.os.Bundle;
+import android.util.Log;
+import android.view.View;
+import android.widget.Toast;
+
+import androidx.annotation.NonNull;
+import androidx.appcompat.app.AppCompatActivity;
+
+import java.time.Instant;
+
+import de.dennisguse.opentracks.AbstractActivity;
+import de.dennisguse.opentracks.R;
+import de.dennisguse.opentracks.data.models.Marker;
+import de.dennisguse.opentracks.data.models.Track;
+import de.dennisguse.opentracks.data.models.TrackPoint;
+import de.dennisguse.opentracks.services.TrackRecordingService;
+import de.dennisguse.opentracks.services.TrackRecordingServiceConnection;
+import de.dennisguse.opentracks.settings.PreferencesUtils;
+import de.dennisguse.opentracks.ui.markers.MarkerEditActivity;
+import de.dennisguse.opentracks.util.IntentUtils;
+
+/**
+ * Public API to creates a Marker for a given track with a given location
+ */
+public class CreateMarkerActivity extends AppCompatActivity {
+
+    public static final String EXTRA_TRACK_ID = "track_id";
+    public static final String EXTRA_LOCATION = "location";
+
+    private static final String TAG = CreateMarkerActivity.class.getSimpleName();
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+
+        if (!PreferencesUtils.isPublicAPIenabled()) {
```

Comment:
This check is rather annoying.
I'm not sure, if it really makes it more secure to check if the public API is enabled.
If the user clicks `cancel` the marker will be deleted anyhow.
@dennisguse what do you think?

Reply:
Not sure. The problem nowadays is that the marker is saved first and deleted only if the user cancels this operation. If the app gets killed the marker will remain.

Reply:
Sounds like a good plan :)

Reply:
Changed the behavior of the `MarkerEditActivity` to insert the Marker at the very end `onDone` in `MarkerEditViewModel`.